test-local/cli-doctor-collect-only.test.js

```diff
@@ -0,0 +1,35 @@
+'use strict'
+
+const fs = require('fs')
+const path = require('path')
+const test = require('tap').test
+const cli = require('../test/cli.js')
+
+test('clinic doctor --collect-only --dest \'./foo\' - no issues', function (t) {
+  cli({ relayStderr: false }, [
+    'clinic', 'doctor', '--collect-only', '--dest', './foo',
+    '--', 'node', '-e', 'setTimeout(() => {}, 300)'
+  ], function (err, stdout, stderr, tempdir) {
+    t.ifError(err)
+    const dirname = stdout.match(/(\d+\.clinic.doctor)/)[1]
+    fs.access(path.resolve(tempdir, 'foo', dirname), function (err) {
+      t.ifError(err)
+      fs.access(path.resolve(tempdir, 'foo', dirname + '.html'), function (err) {
+        t.strictEqual(err.code, 'ENOENT')
+        t.end()
+      })
+    })
+  })
+})
+
+test('clinic doctor --collect-only - bad status code', function (t) {
+  cli({ relayStderr: false }, [
+    'clinic', 'doctor', '--collect-only', '--dest', './foo',
+    '--', 'node', '-e', 'process.exit(1)'
+  ], function (err, stdout, stderr) {
+    t.strictDeepEqual(err, new Error('process exited with exit code 1'))
+    t.strictEqual(stdout, 'To stop data collection press: Ctrl + C\n')
+    t.ok(stderr.includes('process exited with exit code 1'))
```

Comment:
Suggestion also add an assertion that verifies that no clinic folder has been added to the directory due to the failure.

Reply:
This is more complicated as the `stderr` does not expose the target folder or any information about the pid.

Reply:
Hm, true. We could just read the directory but it's probably fine as it is.

Reply:
(We could work around this by reading the folders before and after and only compare the length of all folders but it should not be that important).